_pages/clients/python.md

```diff
@@ -64,206 +60,164 @@ client = Algorithmia.client("YOUR_API_KEY", "https://mylocalendpoint")
-nlp_directory.get_permissions().read_acl == AclType.private # True
-{% endhighlight %}
+You might have noticed that in this example we included a version number when instantiating the algorithm. Pinning your code to a specific version of the algorithm can be especially important in a production environment where the underlying implementation might change from version to version.
 
-Notice that we changed our data collection to private, which means that only you will be able to read and write to your data collection.
+### Request Options
 
-Note that read access that is set to the default `DataMyAlgorithms` allows any algorithm you call to have access to your data collection so most often, this is the setting you want when you are calling an algorithm and are an application developer.
+The client exposes options that can configure algorithm requests. This includes support for changing the timeout or indicating that the API should include stdout in the response. In the following example, we set the timeout to 60 seconds and disabling `stdout` in the response:
 
-For more information on collection-based Access Control Lists (ACLs) and other data collection permissions go to the [Hosted Data Guide]({{site.baseurl}}/data/hosted).
+{% highlight python %}
+algo.set_options(timeout=60, stdout=False)
+{% endhighlight %}
 
-### Upload Data to your Data Collection
+You can find more details in [API Docs](/developers/api/?python) > [Invoke an Algorithm](/developers/api/?python#invoke-an-algorithm).
 
-So far you've created your data collection and checked and updated directory permissions. Now you're ready to upload the text file that you created at the beginning of the guide to your data collection using the Data API.
+### Error Handling
 
-First create a variable that holds the path to your data collection and the text file you will be uploading:
+To be able to better develop across languages, Algorithmia has created a set of standardized errors that can be returned by either the platform or by the algorithm being run. In Python, API errors and Algorithm exceptions will result in calls to pipe throwing an `AlgoException`:
 
 {% highlight python %}
-text_file = "data://YOUR_USERNAME/nlp_directory/jack_london.txt"
+client.algo('util/whoopsWrongAlgo').pipe('Hello, world!')
+# Algorithmia.algo_response.AlgoException: algorithm algo://util/whoopsWrongAlgo not found
 {% endhighlight %}
 
-Next upload your local file to the data collection using the `.putFile()` method:
+You can read more about [Error Handling](/developers/algorithm-development/algorithm-errors) in the [Algorithm Development](/developers/algorithm-development) section of the dev center.
 
-{% highlight python %}
-if client.file(text_file).exists() is False:
-    # Upload local file
-    client.file(text_file).putFile("/your_local_path_to_file/jack_london.txt")
-{% endhighlight %}
+Working with the Algorithmia Data Portal
+----------------------------------------
```

Comment:
nit: update to `## Working with the Algorithmia Data Portal` to match the syntax in the rest of the file.